### PR TITLE
Added back Chromium implementation of views to look for individual cookies stored per-website.

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/27150

Added back Chromium implementation of views to look for individual cookies stored per-website.

https://user-images.githubusercontent.com/2965009/228226221-1b991bba-53e4-442c-b37a-aa3017390423.mp4

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
- Go to `brave://settings/cookies/detail?site=[domain]` and check cookies for this site can be reviewd and deleted
- Check cookies editing page can be navigated from `brave://settings/content/siteDetails` by clicking to `Cookies and Site Data` link

### CODE REVIEW

File: browser/resources/settings/brave_site_settings/site_data_details_subpage.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  indexed_db: loadTimeData.getString('cookieDatabaseStorage'),
+  local_storage: loadTimeData.getString('cookieLocalStorage'),
+  service_worker: loadTimeData.getString('cookieServiceWorker'),
+  shared_worker: loadTimeData.getString('cookieSharedWorker'),
+  quota: loadTimeData.getString('cookieQuotaStorage'),
+};
+
+/**
+ * 'site-data-details-subpage' Display cookie contents.
+ */
```
Comment (petemill) [inline browser/resources/settings/brave_site_settings/site_data_details_subpage.ts:47]: How about a link to where and when we took it from Chromium, so future maintainer has context?

File: browser/resources/settings/sources.gni [hunk 2]
```diff
@@ -58,6 +59,8 @@ brave_settings_non_web_component_files = [
   "brave_appearance_page/brave_appearance_browser_proxy.ts",
   "brave_clear_browsing_data_dialog/brave_clear_browsing_data_dialog_behavior.ts",
   "brave_add_site_dialog/brave_add_site_dialog.ts",
+  "brave_site_details/brave_site_details.ts",
+  "brave_privacy_page/brave_privacy_page.ts",
```
Comment (petemill) [inline browser/resources/settings/sources.gni:63]: please locate these more alphabetically below with your other items

File: browser/ui/webui/settings/settings_cookies_view_handler.cc [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+#include "base/ranges/algorithm.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/values.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/webui/cookies_tree_model_util.h"
+#include "chrome/grit/generated_resources.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/web_ui.h"
+
+namespace settings {
```
Comment (petemill) [inline browser/ui/webui/settings/settings_cookies_view_handler.cc:29]: Again, perhaps provide a link from chromium source where this was retrieved in order to add context to future maintainers. Perhaps also note what had to be modified, if anything.

**Review by petemill — CHANGES_REQUESTED** [review]
Just a couple small changes I think would be needed. That's if we want to take on this maintenance from chromium!

File: browser/resources/settings/brave_site_settings/site_data_details_subpage.ts [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+const SiteDataDetailsSubpageElementBase =
+    RouteObserverMixin(WebUiListenerMixin(PolymerElement)) as {
+      new (): PolymerElement & WebUiListenerMixinInterface &
+          RouteObserverMixinInterface,
+    };
+
+export class SiteDataDetailsSubpageElement extends
+    SiteDataDetailsSubpageElementBase {
+  static get is() {
+    return 'site-data-details-subpage';
```
[the remark this answers is not in the document]
Comment (spylogsster) [reply browser/resources/settings/brave_site_settings/site_data_details_subpage.ts]: done

File: app/settings_strings.grdp [hunk 5]
```diff
@@ -4755,4 +4755,16 @@
...
+  </message>
+  <message name="IDS_SETTINGS_SITE_COOKIES_LINK_DESC" desc="The description of a link to navigate to site cookies subpage.">
+    Show all cookies and locally stored data for this site
+  </message>
+  <message name="IDS_SETTINGS_SITE_SETTINGS_COOKIE_SUBPAGE" desc="A label for cookie subpage, stating which site we're showing data for.">
+    <ph name="SITE">$1<ex>www.example.com</ex></ph> locally stored data
+  </message>
+  <message name="IDS_SETTINGS_SITE_SETTINGS_COOKIES_REMOVE_ALL" desc="A label for remove all cookies button.">
+    Remove All
+  </message>
```
[the remark this answers is not in the document]
Comment (spylogsster) [reply app/settings_strings.grdp]: done

File: browser/resources/settings/brave_site_settings/site_data_details_subpage.ts (hunk 1 above)
Comment (spylogsster) [reply browser/resources/settings/brave_site_settings/site_data_details_subpage.ts]: done

File: browser/resources/settings/sources.gni (hunk 2 above)
Comment (spylogsster) [reply browser/resources/settings/sources.gni]: done

File: browser/ui/webui/settings/settings_cookies_view_handler.cc (hunk 3 above)
Comment (spylogsster) [reply browser/ui/webui/settings/settings_cookies_view_handler.cc:36]: done

**Review by iefremov — APPROVED** [review]
patches LGTM

**Review by petemill — APPROVED** [review]
👍

File: browser/ui/webui/settings/settings_cookies_view_handler_unittest.cc [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+  content::TestWebUI* web_ui() { return web_ui_.get(); }
+  CookiesViewHandler* handler() { return handler_.get(); }
+
+ private:
+  std::unique_ptr<content::TestWebUI> web_ui_;
+  std::unique_ptr<CookiesViewHandler> handler_;
+
+  // Ref pointers to storage helpers used in the tree model used for testing.
+  // Retained to allow control over batch update completion.
+  scoped_refptr<browsing_data::MockLocalStorageHelper>
```
Comment (cdesouza-chromium) [inline browser/ui/webui/settings/settings_cookies_view_handler_unittest.cc:127]: These variables are not named according to convention required by the the style guide [1]. I'm flagging this to raise some awareness, to authors and reviewers, as I've come across a few places today where such cases seemed to have got into master.

[1] https://google.github.io/styleguide/cppguide.html#Variable_Names

**Comment by tlongfellow** [comment]
The cookie page has again been removed. Clearly, Brave developers do not want users to have control over individual cookies - why? One reply is that users can navigate to an offending webpage and click the lock icon and deal with cookies that way. HOWEVER (shouting), this is NOT a viable choice for companies such as doubleclick who make their money by distributing ad cookies all over the place, and there are hundreds of companies like doubleclick. Brave developers are completely ignoring the fact that many of us want to see the cookies already accepted by the Brave browser so that we can build effective domain block lists. PUT THE COOKIE BACK, PLEASE!!!!!

**Pull request merged**